### ISSUE TITLE
[SAE] Compact active-set Newton step ignores step_size for manifold coordinates

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#596]
**Verified — CONFIRMED.**

In `apply_newton_step_impl` (`src/terms/sae_manifold.rs`), the **compact** active-set branch writes coordinate deltas without the line-search `step_size` factor, while logits and the **dense** branch both apply it.

Compact branch (~`src/terms/sae_manifold.rs:4589-4594`):
```rust
delta_coord[row * d + axis] = full_delta[row_base + axis];   // NO step_size
```
Logits in the same branch (~4581) do scale: `step_size * full_delta[...]`.

Dense branch (~4621-4626):
```rust
delta_coord[row * d + axis] = step_size * delta_ext_coord[row_base + axis];   // WITH step_size
```

**Why it's wrong:** a damped Newton/Armijo step must apply θ_new = retract(θ_old, α·Δθ) with α = step_size uniformly. The compact path damps logits/β but takes a full-size coordinate move, so line-search accept/reject decisions are made on a step that is not the actual scaled Newton step. Compact layout is the path used for JumpReLU and large-K sparse plans, so this is on the important paths.

**Fix:** `delta_coord[row*d+axis] = step_size * full_delta[row_base+axis];` and add a regression test: with step_size=0.25, compact coordinate displacement norm must be 0.25× the full Newton displacement (matching dense).

_Source: static SAE/manifold-SAE audit; verified against current code at commit 3c1a4efa2._

**Comment by SauersML** [comment]
Fixed on main in `484cb83f3`: the compact active-set branch of `apply_newton_step_impl` now writes `delta_coord[row*d+axis] = step_size * full_delta[row_base+axis]` (sae_manifold.rs:4644), matching the dense branch.
